### PR TITLE
feat: add router

### PR DESCRIPTION
**Pull request by RationalRyan11**
# Pull Request

## Scope
Added in simple routing.
https://dev.azure.com/JHNMAT015/ECO5037S%20Front%20end/_backlogs/backlog/ECO5037S%20Front%20end%20Team/Backlog%20items/?workitem=81

## Work Done
Implemented a vue router and currently works with "Hello World" component. 

## Steps to Test
```npm run start```
Relevant components should still render.

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    component: HomePage,
+  },
+]
+
+const router = createRouter({
+  history: createWebHistory(),
+  routes,
+})
+
+export default router
```
Comment (sjwu1235) [inline src/router.js:17]: resolve linting issue

File: src/main.js [hunk 2]
```diff
@@ -1,4 +1,9 @@
 import { createApp } from 'vue'
 import App from './App.vue'
+import router from './router'
+
+const app = createApp(App)
+app.use(router)
+app.mount('#app')
+
```
Comment (sjwu1235) [inline src/main.js:9]: delete the 2 empty lines

File: src/App.vue [hunk 3]
```diff
@@ -3,20 +3,18 @@
...
-    <home-page />
+    <router-view />
     <footer>
       <h1>Placeholder for footer</h1>
     </footer>
   </div>
 </template>
 
 <script>
-import HomePage from './components/Home.vue'
```
Comment (sjwu1235) [inline src/App.vue:14]: delete extra line

**Review by sjwu1235 — CHANGES_REQUESTED** [review]
Please remove extra empty lines

File: src/router.js (hunk 1 above)
Comment (RationalRyan11) [reply src/router.js]: What was the linting issue here? I think it auto fixed it but curious

**Review by sjwu1235 — DISMISSED** [review]

File: src/router.js (hunk 1 above)
Comment (sjwu1235) [reply src/router.js]: you need to add a line after 17

Comment (RationalRyan11) [reply src/router.js]: Ah cool, thanks :)

**Review by sjwu1235 — APPROVED** [review]

**Pull request merged**